cmd/agent/entrypoint.go

```diff
@@ -249,6 +257,68 @@ func (ep *Entrypoint) getReporterMetrics() map[string]interface{} {
 	return map[string]interface{}{"enabled-features": ep.cfg.EnabledFeatures}
 }
 
+func (ep *Entrypoint) supportHandler(rw http.ResponseWriter, r *http.Request) {
+	ep.mut.Lock()
+	cfg := ep.cfg
+	ep.mut.Unlock()
+
+	if cfg.DisableSupportBundle {
+		rw.WriteHeader(http.StatusNotFound)
+		_, _ = rw.Write([]byte("404 - support bundle generation is disabled"))
```

Comment:
Is 404 appropriate for this? The endpoint was valid, just denied. Another 4xx code (like `405 Method Not Allowed`) might be appropriate here.

Reply:
We do the same for when we the /-/config endpoint is disabled, but you're right. I feel like 403 Forbidden might make sense here, WDYT?

Reply:
Yeah, 403 seems better than 405 👍